### PR TITLE
Edit inflow outflow error handling

### LINKED ISSUE #13: Track products
As a growing business owner, I want to monitor and edit different quantities of products within StockPal so I can view which products are running out fast and replenish them accordingly.

### PR DESCRIPTION
**Pull request by leongxingyu**

### CODE REVIEW

File: src/main/java/seedu/stockpal/data/product/Product.java [hunk 1]
```diff
@@ -59,15 +60,21 @@ public boolean isPidMatch(Pid pid) {
     }
 
     public void increaseQuantity(Integer amountToChange) {
-        quantity.updateIncreaseQuantity(amountToChange);
+        try {
+            quantity.updateIncreaseQuantity(amountToChange);
+            Ui.printToScreen("Quantity updated. " + quantity.toString());
+        } catch (InventoryQuantityOverflowException IQOE) {
+            Ui.printToScreen("Overflow detected. Not change to quantity. " + quantity.toString());
```
Comment (EdmundTangg) [inline src/main/java/seedu/stockpal/data/product/Product.java]: Typo: supposed to be "No change to quantity."

Comment (leongxingyu) [reply src/main/java/seedu/stockpal/data/product/Product.java]: Updated.

File: src/main/java/seedu/stockpal/data/product/Quantity.java [hunk 2]
```diff
@@ -13,8 +14,12 @@ public Integer getQuantity() {
         return this.quantity;
     }
 
-    public void updateIncreaseQuantity(Integer increaseQuantity) {
-        quantity += increaseQuantity;
+    public void updateIncreaseQuantity(Integer increaseQuantity) throws InventoryQuantityOverflowException {
+        long tentativeQuantity = (long) quantity + (long) increaseQuantity;
+        if (tentativeQuantity >= Integer.MAX_VALUE) {
+            throw new InventoryQuantityOverflowException("Overflow detected. No Change to quantity.");
```
Comment (Kobot7) [inline src/main/java/seedu/stockpal/data/product/Quantity.java:22]: Consider making magic string a constant instead?

File: src/main/java/seedu/stockpal/data/product/Product.java [hunk 3]
```diff
@@ -59,15 +60,21 @@ public boolean isPidMatch(Pid pid) {
...
     }
 
     public void decreaseQuantity(Integer amountToChange) {
         try {
             quantity.updateDecreaseQuantity(amountToChange);
-            Ui.printToScreen("Quantity updated. " + quantity);
-        } catch (InsufficientAmountException e) {
-            Ui.printToScreen("Insufficient amount in inventory. No change to quantity." + quantity);
+            Ui.printToScreen("Quantity updated. " + quantity.toString());
+        } catch (InsufficientAmountException IAE) {
```
Comment (Kobot7) [inline src/main/java/seedu/stockpal/data/product/Product.java:76]: Uppercase variables are usually constants, reconsider if this is applicable here?

File: src/main/java/seedu/stockpal/data/product/Quantity.java (hunk 2 above)
Comment (leongxingyu) [reply src/main/java/seedu/stockpal/data/product/Quantity.java:22]: Edited.

File: src/main/java/seedu/stockpal/data/product/Product.java (hunk 3 above)
Comment (leongxingyu) [reply src/main/java/seedu/stockpal/data/product/Product.java:76]: Noted and edited.

**Comment by wjunjie01** [comment]
LGTM

**Comment by cheeseong2001** [comment]
LGTM

**Pull request merged**